cmake/FindClangTools.cmake

```diff
@@ -0,0 +1,45 @@
+# Try to find clang-tidy and clang-format
+#
+# Usage of this module:
+#
+#  find_package(ClangTools)
+#
+# Variables used by this module, they can change the default behaviour and need
+# to be set before calling find_package:
+#  ClangTools_PATH      When set, this path is inspected instead of standard library binary 
+#                       locations to find clang-tidy and clang-format
+#
+# This module defines:
+#  CLANG_TIDY_BIN       The  path to the clang tidy binary
+#  CLANG_TIDY_FOUND     Whether clang tidy was found
+#  CLANG_FORMAT_BIN     The path to the clang format binary 
+#  CLANG_FORMAT_FOUND   Whether clang format was found
+#
+
+find_program(CLANG_TIDY_BIN
```

Comment:
Do we need this now?

Reply:
I think this is the best way to do this. We push scripts, and the actual formatting is moved to another PR.

Reply:
We should then revert the other files and make a small PR